packages/lib/src/utils/observability/honeycomb.ts

```diff
@@ -0,0 +1,97 @@
+  /** The service name for the app we are running e.g. spruce or parsley */
+  serviceName: string;
+  /** The endpoint for the Honeycomb SDK to send traces to if we are not using the default */
+  endpoint: string;
+  /** The url for our honeycomb instrumented server to connect frontend and backend traces together */
+  backendURL?: string;
+  /** Whether to enable debug mode in the honeycomb sdk this will enable additional logging and print links to traces */
+  debug: boolean;
+  /** The INGEST key for the Honeycomb SDK */
+  ingestKey: string;
+}
+
+/**
+ * Initializes the Honeycomb SDK with the provided configuration.
+ * @param config - The configuration object for the Honeycomb SDK.
+ * @param config.ingestKey - The Honeycomb INGEST API key.
+ * @param config.backendURL - The backend URL.
+ * @param config.debug - Whether to start the SDK in debug mode.
+ * @param config.serviceName - The name of the service.
+ * @param config.endpoint - The endpoint for the Honeycomb SDK to send traces to if we are not using the default.
+ */
+const initializeHoneycomb = ({
+  backendURL,
+  debug,
+  endpoint,
+  ingestKey,
+  serviceName,
+}: HoneycombConfig) => {
+  if (debug && !ingestKey) {
+    console.warn(
+      "Honeycomb INGEST API key was not provided. Starting SDK in debug mode.",
+    );
+    return;
+  }
+  if (!ingestKey) {
+    console.error(
+      "Honeycomb INGEST API key is required to start the SDK in production mode.",
+    );
+  } else {
+    try {
+      const userId = localStorage.getItem("userId") ?? undefined;
+      const honeycombSdk = new HoneycombWebSDK({
+        debug,
+        endpoint,
+        instrumentations: [
+          getWebAutoInstrumentations({
+            "@opentelemetry/instrumentation-fetch": {
+              // Add GraphQL operation name as an attribute to HTTP traces.
+              applyCustomAttributesOnSpan: (span, request) => {
+                if (span && request) {
+                  const body = request.body as string;
+                  const graphqlQuery = detectGraphqlQuery(body);
+                  if (graphqlQuery) {
+                    span.setAttribute(
+                      "graphql.operation_name",
+                      graphqlQuery.operationName,
+                    );
+                    span.setAttribute(
+                      "graphql.query_type",
+                      graphqlQuery.queryType,
+                    );
+                  }
+                }
+              },
+
+              // Allow connecting frontend & backend traces.
+              propagateTraceHeaderCorsUrls: [backendURL || ""],
+            },
+            "@opentelemetry/instrumentation-document-load": {
+              ignoreNetworkEvents: true,
+            },
+          }),
+        ],
+        // Add user.id as an attribute to all traces.
+        resourceAttributes: {
+          "user.id": userId,
+        },
+        localVisualizations: debug,
+        serviceName,
+        apiKey: ingestKey,
```

Comment:
could we add a comment about wanting to move away from the ingest key at some point, maybe with a ticket number?

Reply:
I don't think we do want to move away from the ingestKey its the recommended way to authenticate requests with honeycomb. https://www.honeycomb.io/blog/ingest-only-api-keys Is there something else we should be using?

Reply:
Honeycomb states that ingest key [is not as secure](https://docs.honeycomb.io/send-data/javascript-browser/honeycomb-distribution/#configuration-options-for-browser-telemetry) as sending to a collector, which doesn't require an ingest key. That's what I assumed your conversations with Jonathan were about, having the collector handle the auth.

Reply:
I talked to Jonathan and we managed to get the collector working if a user is on the vpn

Reply:
that sounds fine given that we have WARP which should automatically turn on after a period of time even if disabled by the user